packages/marketplace/contracts/TransferManager.sol

```diff
@@ -199,18 +211,107 @@ abstract contract TransferManager is Initializable, ITransferManager {
         DealSide memory paymentSide,
         DealSide memory nftSide
     ) internal returns (uint256) {
-        (address token, uint256 tokenId) = LibAsset.decodeToken(nftSide.asset.assetType);
-        IRoyaltiesProvider.Part[] memory royalties = royaltiesRegistry.getRoyalties(token, tokenId);
+        if (nftSide.asset.assetType.assetClass == LibAsset.AssetClass.BUNDLE) {
+            LibAsset.Bundle memory bundle = LibAsset.decodeBundle(nftSide.asset.assetType);
+
+            for (uint256 i; i < bundle.bundledERC721.length; i++) {
+                address token = bundle.bundledERC721[i].erc721Address;
+                uint256 idLength = bundle.bundledERC721[i].ids.length;
+                for (uint256 j; j < idLength; j++) {
+                    uint256 tokenId = bundle.bundledERC721[i].ids[j];
+                    IRoyaltiesProvider.Part[] memory royalties = royaltiesRegistry.getRoyalties(token, tokenId);
+
+                    remainder = _applyRoyalties(
+                        remainder,
+                        paymentSide,
+                        nftSide.asset.priceDistribution.erc721Prices[i][j],
+                        royalties,
+                        nftSide.recipient
+                    );
+                }
+            }
+
+            for (uint256 i; i < bundle.bundledERC1155.length; i++) {
+                address token = bundle.bundledERC1155[i].erc1155Address;
+                uint256 idLength = bundle.bundledERC1155[i].ids.length;
+                require(idLength == bundle.bundledERC1155[i].supplies.length, "ERC1155 array error");
+                for (uint256 j; j < idLength; j++) {
+                    uint256 tokenId = bundle.bundledERC1155[i].ids[j];
+                    IRoyaltiesProvider.Part[] memory royalties = royaltiesRegistry.getRoyalties(token, tokenId);
+                    for (uint256 k; k < royalties.length; k++) {
+                        royalties[i].basisPoints = royalties[i].basisPoints * bundle.bundledERC1155[i].supplies[j]; // cumulative basis point
```

Comment:
Can you explain this bit?
If the royalty on an ERC1155 collection is 1% and we purchase 100 copies we will assign it as 100% royalty?

Reply:
For ERC1155 collection if royalty for token id:1 is 1% and a supply of 100 is being sold in Bundle then the cumulative royalty for ERC1155 is 1%*100 
But this condition will revert with ` royalty too high`

Reply:
Royalty for ERC1155 token with id:1 will always be same

Reply:
Doesn't matter how you do it, multiplication is commutative so you can calculate whichever value you want first.

But you do understand the issue with the current code right?

Reply:
yeah I'll update the logic